ros2/rust/string_length_node/src/main.rs

```diff
@@ -0,0 +1,69 @@
+// Rust imports
+use std::sync::{Arc, Mutex};
+// rclrs related imports
+use rclrs::{Publisher, RclrsError};
+use std_msgs::msg::String as StringMsg;
+use std_msgs::msg::UInt8 as UInt8Msg;
+
+// Define the struct for the node
+struct StringLengthNode {
+    node: Arc<rclrs::Node>,
+    _subscription: Arc<rclrs::Subscription<StringMsg>>,
+    data: Arc<Mutex<Option<StringMsg>>>,
+    publisher: Arc<rclrs::Publisher<UInt8Msg>>,
```

Comment:
You have an Arc on everything in here. It would be better to remove the Arcs inside the struct, and wrap the entire instance of the struct in a single Arc. You should be able to transform the program so that works.

Reply:
So I've been investigating and I think that currently the Rust library for ROS does not support this. For example, the function *create_publisher* in the node struct returns an Arc to a publisher.
I've been trying to remove the Arc in each element, but the code fails to compile due to this.

```
pub fn create_publisher<T>(
        &self,
        topic: &str,
        qos: QoSProfile,
    ) -> Result<Arc<Publisher<T>>, RclrsError>
...
```

Reply:
Got it, you'll have to keep at least some of those Arcs then when it comes from the library.